robotoff/app/core.py

```diff
@@ -121,6 +122,9 @@ def get_insights(
     if offset is not None and order_by != "random":
         query = query.offset(offset)
 
+    if group_by:
+        query = query.group_by(keep_types)
```

Comment:
No you should see it's not ok because keep_types is a list of string while group_by expects a peewee field !

Here we want to group by values (value_tag precisely) that what the issue says.

So it's
```suggestion
        query = query.group_by(ProductInsight.value_tag)
```

But as soon as you do this you need to specify how you will aggregate the other fields (because they have different value), and really in our case we are just interested in counting elements. See [peewee documentation](https://docs.peewee-orm.com/en/latest/peewee/querying.html#aggregating-records) but in our case we do not group on one table versus a join table, but we really are interested only in two values: the value_tag and the count of entries, you can manage that using [right parameters to select and scalar as_tupe=True](https://docs.peewee-orm.com/en/latest/peewee/querying.html#retrieving-scalar-values)

The best to find the right query is to set a pdb.set_trace() in one of your test where you have some insight created and see what works.